discgen/utils.py

```diff
@@ -117,6 +124,79 @@ def create_svhn_streams(training_batch_size, monitoring_batch_size):
                           monitoring_batch_size)
 
 
+class Colorize(AgnosticSourcewiseTransformer):
+    def __init__(self, data_stream, **kwargs):
+        super(Colorize, self).__init__(
+             data_stream=data_stream,
+             produces_examples=data_stream.produces_examples,
+             **kwargs)
+
+    def transform_any_source(self, source, _):
+        iters, dim, height, width = source.shape
+        return np.tile(source.reshape(iters*dim, 1, 64, 64), (1, 3, 1, 1))
+
+
+def create_custom_streams(filename, training_batch_size, monitoring_batch_size,
+                          include_targets=False, color_convert=False, allowed=None):
+    """Creates data streams from fuel hdf5 file. Currently features must be 64x64.
+
+    Parameters
+    ----------
+    filename : string
+        basename to hdf5 file for input
+    training_batch_size : int
+        Batch size for training.
+    monitoring_batch_size : int
+        Batch size for monitoring.
+    include_targets : bool
+        If ``True``, use both features and targets. If ``False``, use
+        features only.
+    color_convert : bool
+        If ``True``, input is assumed to be one-channel, and so will
+        be transformed to three-channel by duplication.
+
+    Returns
+    -------
+    rval : tuple of data streams
+        Data streams for the main loop, the training set monitor,
+        the validation set monitor and the test set monitor.
+
+    """
+    img_size = (64, 64)
+    sources = ('features', 'targets') if include_targets else ('features',)
+
+    file_found = False
+    cur_path_index = 0
+    while not file_found and cur_path_index < len(fuel.config.data_path):
+        data_path = fuel.config.data_path[cur_path_index]
+        dataset_fname = os.path.join(data_path, filename+'.hdf5')
+        if os.path.isfile(dataset_fname):
+            file_found = True
+        cur_path_index = cur_path_index + 1
+    if not file_found:
+        print "data set {} not found, exiting".format(dataset_fname)
+        sys.exit(1)
```

Comment:
I think the above block of code can be replaced by a call to `fuel.utils.find_in_data_path`.